src/libraries/trading/PlaceOrder.sol

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+// Bid / Ask actions: puts orders on the book
+// price is always in 10**18 fxp, so a price of 1 is 1 WEI.
+// amount is the number of attoshares the order is for (either to buy or to sell). For a currency with 18 decimals [like ether] if you buy 10**18 at a price of 10**18 then that's going to buy you ONE share [10**18 units] at a cost of ONE ETH [10**18 wei]. For a currency with say 9 decimals, if you buy 10**9 at a price of 10**18 that'll also buy you ONE full unit of that currency worth of shares. If you buy 10**9 at a price of 10**17 that'll buy you POINT_ONE full units of that currency worth of shares [so it'll cost you 10**8]. If you buy 10**8 amount at a price of 10**18 you're also effectively paying POINT_ONE units of currency, this time it's just to get you 10x less shares [in other words you're paying 10x more per share].
+// price is the exact price you want to buy/sell at [which may not be the cost, for example to short a binary market it'll cost 1-price, to go long it'll cost price]
+// smallest order value is 10**14 WEI
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/IController.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/trading/Trading.sol';
+import 'ROOT/trading/IOrders.sol';
+
+
+library PlaceOrder {
+    using SafeMathUint256 for uint256;
+
+    uint256 constant minOrderValue = 10**14;
+
+    struct Data {
+        // Contracts
+        IOrders orders;
+        IMarket market;
+
+        // Order
+        address sender;
+        uint8 outcome;
+        Trading.TradeTypes tradeType;
```

Comment:
I wonder if we should move the enum definition for `TradeTypes` into this contract.  It feels like the right place as Bid/Ask are types of orders, and this contract is an in-memory order contract.

Reply:
Ok, and the TradeDirections enums and new `getOrderTradingTypeFromMakerDirection()`/ `getOrderTradingTypeFromTakerDirection()` functions should go here as well?

Reply:
That seems reasonable to me.